server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -566,4 +562,12 @@ public String toString() {
         }
     }
 
+    static <T> void run(ActionListener<T> listener, CheckedConsumer<ActionListener<T>, Exception> action) {
+        try {
+            action.accept(listener);
+        } catch (Exception e) {
+            listener.onFailure(e);
+        }
+    }
```

Comment:
I wonder if that makes sense to have it non static?
```
    public void completeWith(CheckedConsumer<ActionListener<T>, Exception> action) {
        try {
            action.accept(this);
        } catch (Exception e) {
            this.onFailure(e);
        }
    }
```
and usage:
```
listener.completeWith(action)
```

Though I am not sure if this is a better naming or actually simplifies the usage

Reply:
Yeah we could open that debate about most of these utilities. I'll stick with the common pattern for now tho.